### ISSUE TITLE
Port RCT1-specific vehicles to JSON objects

### ISSUE THREAD

**Issue by Gymnasiast** [OpenRCT2/OpenRCT2#10545]
This has been partially discussed in some other issues. Creating a new one to make it easier to find and bring it all together.

RollerCoaster Tycoon 1 contains some vehicles that aren't in RCT2. With the JSON objects, we can now port them. One issue to resolve is what to do if the user does not have RCT1 sprites.

| Vehicle | Plan/Status |
| ------- | ----------- |
|Inverted Roller Coaster Train (2-seater, Giovanola)|Port to Inverted RC only|
|Spinning Cars (Steel Mini)|Port|
|Rocket Cars (Steel Mini)|Mini RC's rocket cars have been made available on Classic Mini|
|Ladybird trains (Steel Mini)|Port to Classic Mini|
|Log trains (Steel Mini)|Port to Classic Mini|
|Colourable Fruity Ices Stall|Make RCT2's version colourable|

**Comment by duncanspumpkin** [comment]
One option would be to make this an open graphics project.

**Comment by X123M3-256** [comment]
>  One issue to resolve is what to do if the user does not have RCT1 sprites.

Shouldn't it behave the same way expansion objects do when you don't have the expansions?

**Comment by Gymnasiast** [comment]
@X123M3-256 It already behaves like that (invisible sprites). I am not sure if that is the way to go, though: >90% of OpenRCT2 players have the expansions (due to them being included with the GOG and Steam versions), while only a minority of people have OpenRCT2 linked to RCT1.

**Comment by IntelOrca** [comment]
Could we perhaps provide a fallback (new json property) RCT2 sprite for each RCT1 object?

**Comment by duncanspumpkin** [comment]
Thats why I suggest making this into a OpenGraphics project. It would solve this issue.

**Comment by Gymnasiast** [comment]
Yes, but considering that is moving quite slowly, it won't be an option for a long time.

I like @IntelOrca 's solution. It means that people with RCT1 will get the proper sprites, while people without it won't suffer.

**Comment by IntelOrca** [comment]
> while people without it won't suffer.

...too much.

**Comment by Gymnasiast** [comment]
@IntelOrca It means that people without RCT1 will be stuck on the level that OpenRCT2 currently provides, but it won't get worse.

**Comment by IntelOrca** [comment]
For some of the coasters, you might want to choose whether you switch all the sprites to something that caters for all required angles and slopes or something closer but has to use incorrect angled sprites for some of the slopes (like diamond heights junior currently does)

**Comment by X123M3-256** [comment]
@Gymnasiast I know OpenRCT2 doesn't include any custom objects at the moment but is that an option? I mean, you could make a custom version of the RCT1 trains and use that in place of the original sprites.

I feel like people who played RCT1 and view RCT1 parks in OpenRCT2 would probably appreciate the option to use original sprites though. Is it currently possible for a JSON object to specify 16 frames per rotation instead of the normal 32? Last time I tried using RCT1 vehicle sprites in OpenRCT2 I had issues with that.

**Comment by IntelOrca** [comment]
@X123M3-256 the json spec for objects allow you to list as many image IDs / filenames or ID ranges as you like so you can put null (invisible) or alternative sprites where necessary in your image list so that you get 32 frames or how ever many is needed.

**Comment by X123M3-256** [comment]
@IntelOrca Yes, but is there a way to tell the game not to use the sprites that aren't provided? What I did before is duplicate the set of 16 sprites to get 32, but that causes the trains to appear misaligned when turning corners. I am wondering if with the new object format there's a way to specify that only 16 frames are provided.

**Comment by IntelOrca** [comment]
I am not sure what you mean, wouldn't you specify the exact sprite you wanted it to show for each rotation frame?

**Comment by X123M3-256** [comment]
@IntelOrca In RCT2, one full rotation of a car has 32 frames. In RCT1, there are only 16 frames per rotation of the car. So, for half the frames you don't have a corresponding RCT1 sprite, and have to pick one of the neighbouring ones. That causes the train to appear slightly misaligned because the game thinks the sprite has one angle but it actually has another. The effect is that it looks like the trains are drifting slightly on turns (especially small turns).

**Comment by IntelOrca** [comment]
Boats are a bit different since they can basically turn on the spot. With trains you will have to accept that they will make smoother turns and use the neighbouring sprite. I don't think it is anything we can particularly change.

**Comment by pizza2004** [comment]
You could add a flag to the object so the game knows to only use the 16 and has the proper rotations for those specific frames so it matches the look of the original RCT1 instead of having to jam incorrect frames into incorrect rotations from RCT2.

**Comment by spacek531** [comment]
To make things easier to import (i.e. require fewer lines in the sprites field of json objects) I think adding a vehicle DrawMode may be helpful for munging the offsets in the vehicle entry and can re-use most of the existing draw code, only adding new functions where absolutely necessary.

**Comment by ZeeMaji** [comment]
All of these are now done as of https://github.com/OpenRCT2/OpenRCT2/pull/26178
